src/Uno.Toolkit.UI/Controls/NavigationBar/NavigationBarPresenter.Properties.cs

```diff
@@ -0,0 +1,28 @@
+﻿#if IS_WINUI
+using Microsoft.UI.Xaml;
+#else
+using Windows.UI.Xaml;
+#endif
+
+namespace Uno.Toolkit.UI
+{
+	partial class NavigationBarPresenter
+	{
+		
+		public static DependencyProperty NavigationBarContentProperty { get; } = DependencyProperty.Register(
+			nameof(NavigationBarContent),
+			typeof(object),
+			typeof(NavigationBarPresenter),
+			new PropertyMetadata(default));
+
+		/// <summary>
+		/// Gets or sets the NavigationBar Content
+		/// </summary>
+		public object NavigationBarContent
```

Comment:
I might be missing something but why is the NavigationBarContent required? Wouldn't the NavigationBarPresenter just use the Content property from the underlying ContentControl?

Reply:
Yeah that's what I was hoping but you end up with a crash on Windows with a message: "Element is already the child of another element" and I'm not sure how to fix it any other way

A similar situation was happening in DrawerControl but I'm not sure of all the details

Reply:
That is not what I mean though: you should be solely chaining template-binding in the xaml to propagate the changes, instead of setting the binding in NavigationBarPresenter.cs.

The situation of DrawerControl is also different too, there Content and DrawerContent are two concepts that are both used. (Think of a table with drawer, you have both the table top and the drawer for place stuff.)
DrawerControl is itself a ContentControl, and for its **direct** control-template, we are implementing with:
```
<ContentPresenter not_win:Content="{TemplateBinding Content}" />
<ContentControl Content="{TemplateBinding DrawerContent}">
	<ContentControl.Template>
		<ControlTemplate TargetType="ContentControl">
			<Border Background="{TemplateBinding Background}">
				<ContentPresenter not_win:Content="{TemplateBinding Content}" />
			</Border>
		</ControlTemplate>
	</ContentControl.Template>
</ContentControl>
```
note the not_win to avoid doubly "add child": // `not_win:` is applied on the **bolded part**
- Content: at DrawerControl.Content + **at ContentPresenter.Content**
- DrawerContent: at DrawerControl.DrawerContent + at (ContentControl.Content -> **ContentPresenter.Content**)
  ^ must set at ContentControl.Content otherwise we have nothing at all

ContentPresenter within a control-template automatically inherits the content from the template-parent of a ContentContent type, hence the cause of "The element is already a child of another element. (or, something like that)"

I'm not sure if the same setup would work in NavBar given that you have only Content that you need to transpose between NavBar and ComBar through the presenter.

Reply:
It seems that what you have work, so I would keep that.
But just drop Presenter::NavBarContent, and bind through the inherited (Presenter as ContentControl)::Content, and see if that works.
If not, just make Presenter inherit from Control and rename NavBarContent to Content, because the inherited ContentControl::Content is just unused, and make a source of confusion in the future.

Reply:
Yeah binding though the ContentControl's Content property leads to the same error: Element already a child...etc.

I'll switch it back to a Control and rename the property Content.

But I'd still like to understand your other comment, I'm having a hard time grasping what's the issue here. I'll ping you after.